src/libostree/ostree-repo-refs.c

```diff
@@ -216,8 +217,8 @@ resolve_refspec_fallback (OstreeRepo     *self,
 
   if (self->parent_repo)
     {
-      if (!resolve_refspec (self->parent_repo, remote, ref,
-                            allow_noent, &ret_rev, error))
+      if (!resolve_refspec (self->parent_repo, remote, ref, allow_noent,
+                            TRUE, &ret_rev, error))
```

Comment:
Shouldn't this lookup also inherit `fallback_remote`? Otherwise a ref can be found in a parent repo's remotes dir even though you originally specified `FALSE`.